org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java

```diff
@@ -475,7 +475,22 @@ public void didOpen(DidOpenTextDocumentParams params) {
 	@Override
 	public void didChange(DidChangeTextDocumentParams params) {
 		logInfo(">> document/didChange");
-		documentLifeCycleHandler.didChange(params);
+		try {
+			Job[] jobs = Job.getJobManager().find(DocumentLifeCycleHandler.DOCUMENT_LIFE_CYCLE_JOBS);
+			boolean wait = jobs.length > 0;
+			long start = System.currentTimeMillis();
+			while (wait) {
+				jobs = Job.getJobManager().find(DocumentLifeCycleHandler.DOCUMENT_LIFE_CYCLE_JOBS);
+				wait = (System.currentTimeMillis() - start) < 5000 && jobs.length > 0;
+				if (wait) {
+					Thread.sleep(400);
```

Comment:
sleep 100ms

And we should probably move that piece of code to the JobHelper, which itself needs to move from tests to core.